lid.hpp

```diff
@@ -15,19 +16,61 @@ namespace manager
 
 using LidInherit = sdbusplus::server::object_t<
     sdbusplus::xyz::openbmc_project::Software::server::LID>;
+namespace sdbusRule = sdbusplus::bus::match::rules;
 
 class Lid : public LidInherit
 {
 
   public:
-    Lid(sdbusplus::bus_t& bus, const char* path) : LidInherit(bus, path)
+    Lid(sdbusplus::bus_t& bus, const char* path) :
+        LidInherit(bus, path), bus(bus),
+        systemdSignals(
+            bus,
+            sdbusRule::type::signal() + sdbusRule::member("JobRemoved") +
+                sdbusRule::path("/org/freedesktop/systemd1") +
+                sdbusRule::interface("org.freedesktop.systemd1.Manager"),
+            std::bind(std::mem_fn(&Lid::unitStateChange), this,
+                      std::placeholders::_1))
     {}
 
     /**
      * @brief Validator for inband update
      *
      */
     void validate(std::string filePath);
+    /**
+     * @brief Create a image tar ball using lid files provided by host during
+     * inband code update
+     */
+    void assembleCodeUpdateImage();
+
+  private:
+    sdbusplus::bus_t& bus;
+    /** @brief Used to subscribe to dbus systemd signals **/
+    sdbusplus::bus::match_t systemdSignals;
+
+    /**
+     * @brief Check if systemd state change is relevant to this object
+     * Instance specific interface to handle the detected systemd state
+     * change
+     *
+     * @param[in]  msg       - Data associated with subscribed signal
+     *
+     */
+    void unitStateChange(sdbusplus::message_t& msg);
+
+    /** @brief Generate a random number
+     */
+    std::mt19937 randomGen{static_cast<unsigned>(
```

Comment:
any particular reason to use this method?

Reply:
We are supposed to create unique VersionIds for every update. std::mt19937(since C++11) class is a very efficient pseudo-random number generator, so using it with seed as epoch time.